src/Particular.Analyzers/DateTimeOffsetAnalyzer.cs

```diff
@@ -0,0 +1,184 @@
+﻿namespace Particular.Analyzers.Cancellation
+{
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Particular.Analyzers.Extensions;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class DateTimeOffsetAnalyzer : DiagnosticAnalyzer
```

Comment:
Nitpick: this isn't just about assignments any more. It analyzes declarations (which are a kind of assignment I guess), method return statements, and the arguments passed to method invocations. It's really about implicit casts. Maybe

```suggestion
    public class DateTimeImplicitCastAnalyzer : DiagnosticAnalyzer
```